src/components/ChannelPreview/ChannelPreviewMessenger.js

```diff
@@ -63,104 +64,68 @@ const Message = styled.Text`
-                ? formatLatestMessageDate(
-                    this.props.latestMessage.messageObject.created_at,
-                  )
-                : this.props.latestMessage.created_at}
-            </Date>
-          </DetailsTop>
-          <Message
-            unread={this.props.unread > 0 ? this.props.unread : undefined}
-          >
-            {!this.props.latestMessage
-              ? t('Nothing yet...')
-              : truncate(this.props.latestMessage.text.replace(/\n/g, ' '), {
-                  length: this.props.latestMessageLength,
-                })}
-          </Message>
-        </Details>
-      </Container>
-    );
-  }
-}
-
-export default withTranslationContext(themed(ChannelPreviewMessenger));
+const ChannelPreviewMessenger = ({
+  channel,
+  latestMessage,
+  unread,
+  latestMessageLength = 30,
+  formatLatestMessageDate,
+  setActiveChannel,
+}) => {
+  const displayName = useChannelPreviewDisplayName(channel);
+  const displayAvatar = useChannelPreviewDisplayAvatar(channel);
+
+  return (
+    <Container
+      onPress={setActiveChannel.bind(null, channel)}
+      testID="channel-preview-button"
+    >
+      <Avatar size={40} image={displayAvatar.image} name={displayAvatar.name} />
+      <Details>
+        <DetailsTop>
+          <Title ellipsizeMode="tail" numberOfLines={1}>
+            {displayName}
+          </Title>
+          <Date>
+            {formatLatestMessageDate
+              ? formatLatestMessageDate(latestMessage.messageObject.created_at)
+              : latestMessage.created_at}
+          </Date>
+        </DetailsTop>
+        <Message unread={unread > 0 ? unread : undefined}>
+          {latestMessage?.text &&
+            truncate(latestMessage.text.replace(/\n/g, ' '), {
+              length: latestMessageLength,
+            })}
+        </Message>
+      </Details>
+    </Container>
+  );
+};
+
+ChannelPreviewMessenger.propTypes = {
+  /** @see See [Chat Context](https://getstream.github.io/stream-chat-react-native/#chatcontext) */
+  setActiveChannel: PropTypes.func,
+  /** @see See [Chat Context](https://getstream.github.io/stream-chat-react-native/#chatcontext) */
+  channel: PropTypes.object,
+  /** Latest message (object) on channel */
+  latestMessage: PropTypes.object,
+  /** Number of unread messages on channel */
+  unread: PropTypes.number,
+  /** Length at which latest message should be truncated */
+  latestMessageLength: PropTypes.number,
+  /**
+   * Formatter function for date of latest message.
+   * @param date Message date
+   * @returns Formatted date string
+   *
+   * By default today's date is shown in 'HH:mm A' format and other dates
+   * are displayed in 'DD/MM/YY' format. props.latestMessage.created_at is the
+   * default formated date. This default logic is part of ChannelPreview component.
+   */
+  formatLatestMessageDate: PropTypes.func,
+};
+
+ChannelPreviewMessenger.themePath = 'channelPreview';
+
+export default themed(ChannelPreviewMessenger);
```

Comment:
should we put a TODO down for the places `themed` is used to switch this HOC to a themed context eventually?